test/RestrictedNORI.t.sol

```diff
@@ -4,6 +4,469 @@ pragma solidity =0.8.17;
+    uint256 claimableForOriginalSupplier = _rNori
+      .claimableBalanceForScheduleForAccount(
+        scheduleId,
+        _namedAccounts.supplier
+      );
+    assertEq(claimableForOriginalSupplier, 0);
+  }
+}
+
+/** @dev Test correct claimable balance after claiming claimable then transferring remaining tokens, multiple holders */
+contract RestrictedNORI_withdrawFromSchedule_claimableBalancePostTransferMultipleHolders is
+  UpgradeableMarket
+{
+  uint256[] removalIds;
+  uint256 scheduleId = 1;
+  uint256 removalId;
+  uint256 removalId2;
+  uint256 removalId3;
+
+  using UInt256ArrayLib for uint256[];
+
+  function setUp() external {
+    DecodedRemovalIdV0[] memory ids = new DecodedRemovalIdV0[](1);
+    ids[0] = DecodedRemovalIdV0({
+      idVersion: 0,
+      methodology: 1,
+      methodologyVersion: 0,
+      vintage: 2018,
+      country: "US",
+      subdivision: "IA",
+      supplierAddress: _namedAccounts.supplier,
+      subIdentifier: _REMOVAL_FIXTURES[0].subIdentifier + 1
+    });
+    removalId = RemovalIdLib.createRemovalId(ids[0]);
+
+    _removal.grantRole(_removal.CONSIGNOR_ROLE(), address(this));
+    _removal.mintBatch({
+      to: _namedAccounts.supplier,
+      amounts: new uint256[](1).fill(1 ether),
+      removals: ids,
+      projectId: scheduleId,
+      scheduleStartTime: block.timestamp,
+      holdbackPercentage: 50
+    });
+
+    DecodedRemovalIdV0[] memory ids2 = new DecodedRemovalIdV0[](1);
+    ids2[0] = DecodedRemovalIdV0({
+      idVersion: 0,
+      methodology: 1,
+      methodologyVersion: 0,
+      vintage: 2018,
+      country: "US",
+      subdivision: "IA",
+      supplierAddress: _namedAccounts.supplier2,
+      subIdentifier: _REMOVAL_FIXTURES[0].subIdentifier + 1
+    });
+    removalId2 = RemovalIdLib.createRemovalId(ids2[0]);
+
+    _removal.mintBatch({
+      to: _namedAccounts.supplier2,
+      amounts: new uint256[](1).fill(1 ether),
+      removals: ids2,
+      projectId: scheduleId,
+      scheduleStartTime: block.timestamp,
+      holdbackPercentage: 50
+    });
+
+    DecodedRemovalIdV0[] memory ids3 = new DecodedRemovalIdV0[](1);
+    ids3[0] = DecodedRemovalIdV0({
+      idVersion: 0,
+      methodology: 1,
+      methodologyVersion: 0,
+      vintage: 2018,
+      country: "US",
+      subdivision: "IA",
+      supplierAddress: _namedAccounts.supplier3,
+      subIdentifier: _REMOVAL_FIXTURES[0].subIdentifier + 1
+    });
+    removalId3 = RemovalIdLib.createRemovalId(ids3[0]);
+
+    _removal.mintBatch({
+      to: _namedAccounts.supplier3,
+      amounts: new uint256[](1).fill(1 ether),
+      removals: ids3,
+      projectId: scheduleId,
+      scheduleStartTime: block.timestamp,
+      holdbackPercentage: 50
+    });
+
+    _rNori.grantRole(_rNori.MINTER_ROLE(), address(this));
+    _bpNori.grantRole(_bpNori.DEPOSITOR_ROLE(), address(this));
+    _bpNori.deposit(address(_rNori), abi.encode(1 ether));
+  }
+
+  function testClaimableAmountWithMultipleSuppliers() external {
```

Comment:
this is the test that illustrates the issue that can't be solved with our approach